### PR TITLE
[face-detector] remove deprecated package and usages

### PR DESCRIPTION
**Pull request by Simek**
# Why

I was deprecated since SDK 51 and scheduled for the removal.

# How

Remove package code, related code and usages across the code base.

# Test Plan

Done some basic checks and tests locally, but let's run the CI for comprehensive validation.

### CODE REVIEW

File: apps/expo-go/ios/Exponent.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1537,7 +1537,6 @@
 				"${PODS_CONFIGURATION_BUILD_DIR}/GoogleUtilities/GoogleUtilities_Privacy.bundle",
 				"${PODS_CONFIGURATION_BUILD_DIR}/PromisesObjC/FBLPromises_Privacy.bundle",
 				"${PODS_CONFIGURATION_BUILD_DIR}/PromisesSwift/Promises_Privacy.bundle",
-				"${PODS_CONFIGURATION_BUILD_DIR}/RCT-Folly/RCT-Folly_privacy.bundle",
```
Comment (Kudo) [inline apps/expo-go/ios/Exponent.xcodeproj/project.pbxproj:1540]: this could be unrelated change. please try to run the command to ensure cocoapods doesn't have any outdated cache
```
rm -rf apps/bare-expo/ios/Pods apps/bare-expo/ios/build apps/expo-go/ios/Pods apps/expo-go/ios/build && et pods -f
```

File: apps/expo-go/ios/Podfile.lock [hunk 2]
```diff
@@ -3099,7 +3099,7 @@ CHECKOUT OPTIONS:
 SPEC CHECKSUMS:
   boost: 1dca942403ed9342f98334bf4c3621f011aa7946
   CocoaLumberjack: 2f44e60eb91c176d471fdba43b9e3eae6a721947
-  DoubleConversion: f16ae600a246532c4020132d54af21d0ddb2a385
```
Comment (Kudo) [inline apps/expo-go/ios/Podfile.lock:3102]: also unrelated change. please clean Pods/build to reinstall pods

**Review by Kudo — APPROVED** [review]
neat! thanks much for help removing the package and making our repo healthy.
just the xcode project and podfile.lock could be outdated on your machine. please try to clean pods/build and reinstall pod

**Review by amandeepmittal — APPROVED** [review]
